src/Moq/Obsolete/MockFactory.cs

```diff
@@ -363,8 +363,18 @@ public virtual void VerifyAll()
 		}
 
 		/// <summary>
-		/// Invokes <paramref name="verifyAction"/> for each mock 
-		/// in <see cref="Mocks"/>, and accumulates the resulting 
+		/// Calls <see cref="Mock.VerifyNoOtherCalls"/> on all mocks created by this factory.
+		/// </summary>
+		/// <seealso cref="Mock.VerifyNoOtherCalls"/>
+		/// <exception cref="MockException">One or more mocks had setups that were not satisfied.</exception>
+		public void VerifyNoOtherCalls()
+		{
+			VerifyMocks(Mock.VerifyNoOtherCalls);
+		}
```

Comment:
I would feel more comfortable if you could just implement the logic to call `VerifyNoOtherCalls` and aggregate exceptions right here rather than using `VerifyMocks`. Like you've noticed, `VerifyMocks` as it is is very limited in the way how it aggregates exceptions. It's a pity that method has been made `protected virtual` as it really only works for `Verify` and `VerifyAll`, overriding it is pretty pointless.

I greatly appreciate the effort that you made by trying to make `VerifyMocks` more general (by declaring a new `MockExceptionReason` enumeration value for multiple various verification errors etc.), but I am not sure that it would pay off. First, because Moq's verification error messages might not be precise enough in each case to still make sense when they're blindly concatenated. Second, because `MockException` simply isn't designed to carry several verification errors as a payload, and this cannot be solved cleanly with an additional enumeration value.

I'd prefer if you could leave `VerifyMocks` untouched and do something like this in `VerifyNoOtherCalls` instead:

```csharp
var errors = new List<MockException>();

foreach (var mock in mocks)
{
    try
    {
        Mock.VerifyNoOtherCalls(mock);
    }
    catch (MockException error) when (error.Reason == MockExceptionReason.UnverifiedInvocations)
    {
        errors.Add(error);
    }
}

if (errors.Count > 0)
{
    throw MockException.UnverifiedInvocations(errors);
}
```